lib/Dialect/FIRRTL/Transforms/CheckCombCycles.cpp

```diff
@@ -81,6 +157,20 @@ class ChildIterator
     auto isChild = [&]() {
       if (auto connect = dyn_cast<FConnectLike>(childIt->getOwner()))
         return childIt->get() == connect.getSrc();
+      // For a SubIndex/Subfield op, next child is valid, only if it includes
+      // the fieldIndex.
+      if (auto sub = dyn_cast<SubfieldOp>(childIt->getOwner())) {
+        auto fRef = sub.getInput().getType().cast<BundleType>().getFieldID(
+            sub.getFieldIndex());
+        return (fieldIndex < fRef + getBitWidth(sub.getType(), true).value() &&
+                fieldIndex >= fRef);
```

Comment:
I guess getBitwidth and fieldID are irrelevant. Probably `sub.getType().getMaxFieldID()` is what you are looking for?